lib/deaf_grandma.rb

```diff
@@ -17,11 +19,16 @@ def run!
     end
   end
 
-
   def speak(input)
-
-    #Implement your code here <<<<<<<<<
-
+    if input == 'BYE!'
+      @bye_counter += 1
+      if @bye_counter >= 3
+        return 'SEE YOU LATER SONNY!'
+        exit 1 # this isn't right
```

Comment:
Why isn't it right?

Reply:
I know that line 27 is never going to be executed because of the return on line 26. Currently, exit would have to be called from the test spec itself.

Reply:
Right on. I keep seeing submissions that don't address the requirement that the game is supposed to end at this stage, now I think I see why. The `run!` method prints the return value of this method, so when are you supposed to `exit`?

Perhaps [`abort`](http://ruby-doc.org/core-2.4.2/Kernel.html#method-i-abort) helps?

Reply:
Awesome, thanks for the hint!